src/main/java/org/jabref/logic/exporter/EndnoteXmlExporter.java

```diff
@@ -0,0 +1,201 @@
+package org.jabref.logic.exporter;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.jabref.logic.util.StandardFileType;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.types.EntryType;
+import org.jabref.model.entry.types.IEEETranEntryType;
+import org.jabref.model.entry.types.StandardEntryType;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+public class EndnoteXmlExporter extends Exporter {
+    private static final List<EntryType> ORDERED_ENTRY_TYPES = List.of(
+            StandardEntryType.Article, // Journal Article -1
```

Comment:
Why not make an enum or a record all the necessary information?
e.g. EndnoteEntryTypes(StandardEntryType jabrefEntryType, String endnoteEntryType, int order)

Reply:
Initially, I was using the LinkedHashMap approach:
```
private static final Map<EntryType, String> ENTRY_TYPE_MAPPING = new LinkedHashMap<>(Map.ofEntries(
            Map.entry(StandardEntryType.Article, "Journal Article"),
            Map.entry(StandardEntryType.Book, "Book"),
            Map.entry(StandardEntryType.InBook, "Book Section"),
            Map.entry(StandardEntryType.InCollection, "Book Section"),
            Map.entry(StandardEntryType.Proceedings, "Conference Proceedings"),
            Map.entry(StandardEntryType.MastersThesis, "Thesis"),
            Map.entry(StandardEntryType.PhdThesis, "Thesis"),
            Map.entry(StandardEntryType.TechReport, "Report"),
            Map.entry(StandardEntryType.Unpublished, "Manuscript"),
            Map.entry(StandardEntryType.InProceedings, "Conference Paper"),
            Map.entry(StandardEntryType.Conference, "Conference"),
            Map.entry(IEEETranEntryType.Patent, "Patent"),
            Map.entry(StandardEntryType.Online, "Web Page"),
            Map.entry(IEEETranEntryType.Electronic, "Electronic Article"),
            Map.entry(StandardEntryType.Misc, "Generic")
    ));

    private static final Map<EntryType, String> EXPORT_REF_NUMBER = ENTRY_TYPE_MAPPING.keySet().stream().collect(
            Collectors.toMap(entryType -> entryType, entryType -> Integer.toString(ENTRY_TYPE_MAPPING.keySet().stream().toList().indexOf(entryType) + 1), (e1, e2) -> e1, LinkedHashMap::new)); 
           
```

But then no matter what I did, the order of reference type number was random. This was just a workaround that finally fixed it. I'll try your suggestion.